libraries/chain/finality/qc.cpp

```diff
@@ -309,10 +375,54 @@ std::optional<qc_t> aggregating_qc_t::get_best_qc(block_num_type block_num) cons
    return std::optional<qc_t>{qc_t{block_num, std::move(*active_best_qc), {}}};
 }
 
+// A dual finalizer votes on both active and pending finalizer policies
+void aggregating_qc_t::verify_dual_finalizers_votes(const qc_t& qc) const {
```

Comment:
This function seems rather inefficient. I think it would be better if it did sequential scans of the vector instead of creating the sets and maps with all the copies. Alternatively, could sort the finalizers. Seems like we could sort the finalizers when we set them. Going to want them sorted for the diff algorithm also.

Reply:
Initially I was thinking about doing it manually, but ended up with STD containers for simplicity.  Will think about more.

Reply:
Changed to use sorted pending policy keys directly https://github.com/AntelopeIO/spring/pull/475/commits/90643d0213721fa5e1f0cac1b7f1ffe368d5fb0e